Client/Frontend/Browser/TabManager.swift

```diff
@@ -414,25 +414,31 @@ class TabManager: NSObject {
         assert(count == prevCount - 1, "Make sure the tab count was actually removed")
 
         // There's still some time between this and the webView being destroyed. We don't want to pick up any stray events.
+
         tab.webView?.navigationDelegate = nil
 
         if notify {
             delegates.forEach { $0.get()?.tabManager(self, didRemoveTab: tab) }
         }
 
-        if !tab.isPrivate && viableTabs.isEmpty {
-            addTab()
-        }
+        tab.deleteWebView()
 
-        // If the removed tab was selected, find the new tab to select.
-        if selectedTab != nil {
-            selectTab(selectedTab, previous: oldSelectedTab)
-        } else {
-            selectTab(tabs.last, previous: oldSelectedTab)
-        }
+        // Allows the WKWebView instance count to drop before creating another.
+        DispatchQueue.main.async {
```

Comment:
This might make it harder to reason in what order the TabManager delegate methods are called. As long as we correctly deallocate a webview do we need to do this?

Reply:
I mention this at end of https://github.com/mozilla-mobile/firefox-ios/pull/3479#issue-275516570, that is part of an experimental commit to let the webview count drop to zero